<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Gabriel Sanhueza</author>
 <class>QShaderViewer</class>
 <widget class="QMainWindow" name="QShaderViewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QShaderViewer</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../resources/icon.png</normaloff>../resources/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="OGLWidget" name="openGLWidget"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1280</width>
     <height>47</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionLoadGeometry"/>
    <addaction name="actionLoadVertex"/>
    <addaction name="actionLoadFragment"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionTutorial"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionTutorial">
   <property name="text">
    <string>&amp;Tutorial</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="actionLoadGeometry">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Load &amp;geometry file</string>
   </property>
  </action>
  <action name="actionLoadVertex">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Load &amp;vertex shader</string>
   </property>
  </action>
  <action name="actionLoadFragment">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Load &amp;fragment shader</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>src/oglwidget.h</header>
   <slots>
    <slot>receiveModel(Model)</slot>
    <slot>keyPressed(QKeyEvent*)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>QShaderViewer</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadGeometry</sender>
   <signal>triggered()</signal>
   <receiver>QShaderViewer</receiver>
   <slot>loadGeometryClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadVertex</sender>
   <signal>triggered()</signal>
   <receiver>QShaderViewer</receiver>
   <slot>loadVertexClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadFragment</sender>
   <signal>triggered()</signal>
   <receiver>QShaderViewer</receiver>
   <slot>loadFragmentClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>QShaderViewer</sender>
   <signal>emitModel(Model)</signal>
   <receiver>openGLWidget</receiver>
   <slot>receiveModel(Model)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTutorial</sender>
   <signal>triggered()</signal>
   <receiver>QShaderViewer</receiver>
   <slot>loadTutorialClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>QShaderViewer</receiver>
   <slot>loadAboutClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>QShaderViewer</sender>
   <signal>keyPressed(QKeyEvent*)</signal>
   <receiver>openGLWidget</receiver>
   <slot>keyPressed(QKeyEvent*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>keyPressed(QKeyEvent*)</signal>
  <slot>loadTutorialClicked()</slot>
  <slot>loadAboutClicked()</slot>
 </slots>
</ui>
